src/dict.c

```diff
@@ -122,11 +122,13 @@ uint64_t dictGenCaseHashFunction(const unsigned char *buf, size_t len) {
 #define ENTRY_PTR_MASK     7 /* 111 */
 #define ENTRY_PTR_NORMAL   0 /* 000 */
 #define ENTRY_PTR_NO_VALUE 2 /* 010 */
+#define ENTRY_PTR_IS_ODD_KEY  1 /* XX1 : If it is a pointer to odd key address (must be 1). */
+#define ENTRY_PTR_IS_EVEN_KEY 4 /* 100 : If it is a pointer to even key address. */
```

Comment:
any specific reason why 4 but not 2 here? Just asking if I'm missing something.

Reply:
Hope this will be more readable:
```
#define ENTRY_PTR_MASK        7 /* 111 */
#define ENTRY_PTR_NORMAL      0 /* 000 : If a pointer to an entry with value. */
#define ENTRY_PTR_IS_ODD_KEY  1 /* XX1 : If a pointer to odd key address (must be 1). */
#define ENTRY_PTR_IS_EVEN_KEY 2 /* 010 : If a pointer to even key address. (must be 2 or 4). */ 
#define ENTRY_PTR_NO_VALUE    4 /* 100 : If a pointer to an entry without value. */ 
```